libcontainer/factory_linux.go

```diff
@@ -159,13 +159,27 @@ func (l *LinuxFactory) Create(id string, config *configs.Config) (Container, err
 	}
 	containerRoot := filepath.Join(l.Root, id)
 	if _, err := os.Stat(containerRoot); err == nil {
-		return nil, newGenericError(fmt.Errorf("Container with id exists: %v", id), IdInUse)
+		pid, err := ioutil.ReadFile(filepath.Join(containerRoot, "task"))
+		if err == nil {
```

Comment:
Try to indent the error flow and keep the normal flow as unindented as possible.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow